lib/slice.go

```diff
@@ -0,0 +1,107 @@
+package lib
+
+import "fmt"
+
+type Slice[T any] []T
+type Predicate[T any] func (iteratee T) bool
+
+// Filter returns slice items matching specified predicate
+func (slice Slice[T]) Filter(predicate Predicate[T]) Slice[T] {
+	filtered := Slice[T]{}
+	for _, v := range slice {
+		if predicate == nil || predicate(v) {
+			filtered = append(filtered, v)
+		}
+	}
+	return filtered
+}
```

Comment:
```suggestion
func Filter(slice Slice[T], predicate Predicate[T]) Slice[T] {
	filtered := Slice[T]{}
	for _, v := range slice {
		if predicate == nil || predicate(v) {
			filtered = append(filtered, v)
		}
	}
	return filtered
}
```
i think this type of params maybe better for utilization in code

Reply:
Can you be more specific on the reasons why? I aggree that this could have less friction with legacy code, but at this point we should question what is indeed _idiomatic_ in Golang. It seems quite reasonable that generic type abstractions like `Slice[Type]` or `Map[Type]` would be a better choice for 1.18+ Golang code, as custom, reusable behaviour can be easily added to satistfy application needs.

Reply:
i think is better have less friction with  legacy code, cuz the community probably will make something similar to this PR in the std lib, so we can make an agnostic function that we recive any slice is better for now